model/map/src/main/java/org/keycloak/models/map/userSession/MapUserSessionProviderFactory.java

```diff
@@ -64,6 +51,11 @@ public String getHelpText() {
     public void invalidate(KeycloakSession session, InvalidableObjectType type, Object... params) {
         if (type == USER_BEFORE_REMOVE) {
             create(session).removeUserSessions((RealmModel) params[0], (UserModel) params[1]);
+        } else if (type == REALM_BEFORE_REMOVE) {
+            MapUserSessionProvider mapUserSessionProvider = create(session);
+
+            mapUserSessionProvider.removeUserSessions((RealmModel) params[0]);
+            mapUserSessionProvider.removeOfflineUserSessions((RealmModel) params[0]);
```

Comment:
Can this be done using only one storage call?

Reply:
Yes, I changed the method to remove both online and offline sessions.